compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -28,6 +28,8 @@ public type File readonly & object {
     public function lineColumn(Position pos) returns LineColumn;
     public function lineContent(Position|Range range) returns [string, string, string];
     public function qNameRange(Position startPos) returns Range;
+    public function startPos() returns Position;
```

Comment:
I don't see why you need this. Get the start and end position as you parse the ModulePart, same as everything else.

Reply:
The problem is we create the `ModulePart` before we have finished parsing it's `ModuleLevelDefn`s so we don't know the startPos (if it didn't have any imports) and endPos.